src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1132,10 +1133,80 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
     private boolean promptUser(PromptBehavior prompt) {
         return prompt == PromptBehavior.Always || prompt == PromptBehavior.REFRESH_SESSION;
     }
+    
+    /**
+     * check the redirectUri before sending the request
+     * if the redirectUri from the client does not match the valid redirectUri
+     * the client app would not jump to the login page
+     * redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE
+     * 
+     * @param callbackHandle
+     * @param request
+     * @throws UnsupportedEncodingException
+     * @return true if the RedictUri is valid
+     */
+    private boolean verifyBrokerRedirectUri(CallbackHandler callbackHandle, final AuthenticationRequest request){   
+        final String methodName = ":verifyBrokerRedirectUri";
+        String errMsg = "";
+        final String inputUri = request.getRedirectUri();
+        AuthenticationException exception = null;
+        
+        if(StringExtensions.IsNullOrBlank(inputUri))
+        {
+            errMsg = "the redirectUri is null or blank";
+            exception = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+            Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+        }
+        else if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://"))
+        {
+            errMsg = "redirectUri does not start with the valid redict prefix " + AuthenticationConstants.Broker.REDIRECT_PREFIX; 
+            exception = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+            Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+        } 
+        else
+        {
+            try 
+            {
+                PackageHelper packageHelper = new PackageHelper(mContext);
+                String urlEncodedPackagename = URLEncoder.encode(mContext.getPackageName(), AuthenticationConstants.ENCODING_UTF8);
+                String urlEncodedSignature = URLEncoder.encode(packageHelper.getCurrentSignatureForPackage(mContext.getPackageName()), AuthenticationConstants.ENCODING_UTF8);
```

Comment:
@kpanwar nitpick: What is a "synthetic change"?

If we are just talking about renaming something then let's not add yet another issue that will never get done.  Let's just do it.  Unless I am missing something and it is a harder change than it sounds.